### PR TITLE
Adding hook `isVisibleElement`

### PR DESCRIPTION
**Pull request by cliffparnitzky**

### CODE REVIEW

**Comment by cliffparnitzky** [comment]
@aschempp Any suggestions what to fix for a successful CI check?

**Comment by aschempp** [comment]
Unfortunately I need to check this in October (after my holidays).

**Comment by cliffparnitzky** [comment]
okay, thanks.

**Comment by aschempp** [comment]
I have fixed the Travis issue in https://github.com/contao/docs/commit/5bf51bd47d88eb92977ea3308fce0550a26efe13. Apparently the new version of GitBook uses a new version of Nunjucks (template parsing engine) which does not allow for paths outside the book root folder…

**Comment by aschempp** [comment]
Please rebase your PR on the latest master branch

**Comment by cliffparnitzky** [comment]
done

File: api/extensions/hooks/isVisibleElement.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+`getArticle`, `getFrontendModule` and `getContentElement` one can prevent generating
+the complete markup. The hook passes the model of the instance and the current
+visibility state as arguments and expects the new visibility state as return value.
+
+
+## Parameters
+
+1. *Database_Result* `$objElement`
+
+    Database result set from table `tl_content`.
```
Comment (aschempp) [inline api/extensions/hooks/isVisibleElement.md]: it's not always `tl_content`, right?

File: api/extensions/hooks/isVisibleElement.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# isVisibleElement
+
+The `isVisibleElement` hook is triggered when checking if an element should be
+visible in the front end or not. An "element" in this case means either an article,
+a front end module or a content element. In contrast to the other three hooks
+`getArticle`, `getFrontendModule` and `getContentElement` one can prevent generating
+the complete markup. The hook passes the model of the instance and the current
+visibility state as arguments and expects the new visibility state as return value.
+
+
```
Comment (aschempp) [inline api/extensions/hooks/isVisibleElement.md:12]: Please also add the availability (Contao 3.5?) information.

Comment (cliffparnitzky) [inline api/extensions/hooks/isVisibleElement.md:12]: Urgs, missed it, when copying the `isAllowedToEditComment` file.

File: api/extensions/hooks/isVisibleElement.md (hunk 1 above)
Comment (cliffparnitzky) [inline api/extensions/hooks/isVisibleElement.md]: Yep, is corrected.

**Comment by aschempp** [comment]
Great! Thanks for the fixes.

**Pull request merged**